Assignment_1/palindrome.js

```diff
@@ -0,0 +1,46 @@
+class Node {
+    constructor(val) {
+        this.data = val;
+        this.next = null;
+    }
+}
+//
+const palindrome = (head) => {
+    if (!head || !head.next) {
+        return true; 
+    }//empty or one letter
+
+    let fast = head, slow = head;
+    const stack = [];
+
+    while (fast && fast.next) {
+        stack.push(slow.data);
+        fast = fast.next.next;
+        slow = slow.next; 
+    }//push first half of linked list onto stack
+
+    if (fast) { // if list has odd number of elements move slow past middle
+        slow = slow.next;
+    }
+
+    //going to compare stack aka first half with rest aka slow
+    while (slow) {
+        const pop = stack.pop();
+        if (pop !== slow.data) {
+          return false;
+        }
+        slow = slow.next;
+    }
```

Comment:
This could also be written using a for loop, but either way is acceptable.
```suggestion
    for (; slow; pop = stack.pop, slow = slow.next) {
      if (stack.pop() !== slow.data) {
        return false;
      }
    }
```